lib/Net/Etcd/KV.pm

```diff
@@ -59,6 +57,29 @@ sub range {
     return $range
 }
 
+=head2 deleterange
+
+DeleteRange deletes the given range from the key-value store. A delete
+request increments the revision of the key-value store and generates a
+delete event in the event history for every deleted key.
+
+    $etcd->deleterange({key => 'test0'})
+
+=cut
+
+sub deleterange {
```

Comment:
```
     my ( $self, $options ) = @_;
     my $cb = pop if ref $_[-1] eq 'CODE';
     my $range = Net::Etcd::KV::Range->new(
         etcd     => $self,
         cb       => $cb,
         ( $options ? %$options : () ),
     );
     my $delete_range = $range->delete;
     $delete_range->request unless $delete_range->hold;
     return $delete_range;
 }
```
I get where your going here but adding the class adds a lot of repetative code. Lets move this to Etcd.pm while keeping it in the Range class and using existing delete.

Reply:
@hexfusion, thanks for reviewing. I reviewed the swagger spec prior to submitting PR --
 `etcdserverpbRangeRequest` object vs `etcdserverpbDeleteRangeRequest` object and the latter supports just three properties (`key`, `range_end` and `prev_kv`) while the former supports about 13 properties, so the code duplication isn't much and isn't any more than the duplication between `put` and `range` for example.

Also, I just tried to mimic my changes to be as close to the grpc-proxy endpoint as possible. As `deleterange` is an endpoint I can directly call, segregating it from a range object made more sense to me.  Thus, `/kv/deleterange`, `/kv/range`, `/kv/put` now all belongs to KV.pm.

Does that make sense?

Reply:
@foobargeez fair points.  Please update the tests for this as well.  As soon as we can get the travis tests to pass I will put this together for a new release. Thanks again!

Reply:
@hexfusion, thanks.  I apologize for missing tests :-/  I fixed the tests and pushed the changes now.  Please let me know if there is anything pending.